### PR TITLE
Update behavior tree cpp

### PR DESCRIPTION
**Pull request by mjeronimo**
## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | None |
| Primary OS tested on | Ubuntu 18.04 |
| Primary platform tested on | NUC, HP Envy |

--- 

## Description of contribution in a few bullet points

* Update the BehaviorTree.CPP library to the latest version. This version fixes a few issues
    - M_PI macro undefined on Ubuntu 18.04
    - New ParallelNode constructor added so that it can be registered with the bt_factory 
    - ParallelNode is registered in the library so that it can be used in XML
--- 

## Future work that may be required in bullet points

* What remains is to provide a PR to the BehaviorTree repo for the colcon-enabled CMakeLists.txt
* Work with @SteveMacenski to understand how we can incorporate this library into our official build

### CODE REVIEW

**Comment by mjeronimo** [comment]
@SteveMacenski I've been fixing issues with the library and have worked with Davide (the developer on the BehaviorTree.CPP project) to get these changes in the upstream repo. At this point, the only difference that remains is that I've written a CMakeLists.txt file that works with colcon. I've tried using the existing upstream CMakeLists.txt as-is, but couldn't get it to work. So, unless someone can figure that out, my plan is to integrate my changes into the CMakeLists.txt and submit a PR to the upstream project. At that point, we can use that repo instead of my own. Are the options you list the same if we're using the upstream repo?

**Comment by SteveMacenski** [comment]
If we can get this merged back to Davide's repo, then the best option is to have him (or one of us) release this to the ros build farm (i.e. add rosdep keys, prep a bloom release and release in Crystal). Then we just add it as a rosdep key in the package(s) that require it. Given that your changes require colon, it should be merged into a ros2 (or similar) branch to be released on the ros2 track. It then requires Davide to take responsibility to support the ros2 configuration and bloom release to the `ros2/rosdistro` repo along with him melodic track/cherrypicked commits. Assuming he's willing and able to do so, that's the best outcome. 

If not, we can: 1 add as part of the nav2 repo as a lib, 2 `nav2` headed item and release it ourselves, or 3 release your fork to bloom. All of these allow us to retain control over release patterns. 

tl;dr: if Davide's willing to support/maintain the ros2 configs we should have your changes merged into a `ros2` branch and have it released to bloom. If not, we'll need to release ourselves or include in this repository.

**Comment by mjeronimo** [comment]
@SteveMacenski  Now that Davide has integrated my changes to the CMakeLists.txt in his repo, I've updated this PR to point to his repo instead of my local fork. I've created an issue there,

https://github.com/BehaviorTree/BehaviorTree.CPP/issues/28

to discuss next steps on the build farm. Perhaps you could chime in to help resolve. Thanks.

**Comment by SteveMacenski** [comment]
@mjeronimo it sounds like they resolved it, but let me know if I'm misreading something in that chain (I think it was resolved after you commented on this but before I saw).
